src/libraries/System.Text.RegularExpressions/tests/Regex.KnownPattern.Tests.cs

```diff
@@ -137,13 +142,759 @@ string DomainMapper(Match match)
+        {
+            const string Pattern = @"(\b(\w+)\W+)+";
+            const string Input = "This is a short sentence.";
+            Match match = Regex.Match(Input, Pattern, options);
+
+            var actual = new StringBuilder();
+            actual.AppendLine($"Match: '{match.Value}'");
+            for (int ctr = 1; ctr < match.Groups.Count; ctr++)
+            {
+                actual.AppendLine($"   Group {ctr}: '{match.Groups[ctr].Value}'");
+                int capCtr = 0;
+                foreach (Capture capture in match.Groups[ctr].Captures)
+                {
+                    actual.AppendLine($"      Capture {capCtr}: '{capture.Value}'");
+                    capCtr++;
+                }
+            }
+
+            string expected =
+                "Match: 'This is a short sentence.'" + Environment.NewLine +
+                "   Group 1: 'sentence.'" + Environment.NewLine +
+                "      Capture 0: 'This '" + Environment.NewLine +
+                "      Capture 1: 'is '" + Environment.NewLine +
+                "      Capture 2: 'a '" + Environment.NewLine +
+                "      Capture 3: 'short '" + Environment.NewLine +
+                "      Capture 4: 'sentence.'" + Environment.NewLine +
+                "   Group 2: 'sentence'" + Environment.NewLine +
+                "      Capture 0: 'This'" + Environment.NewLine +
+                "      Capture 1: 'is'" + Environment.NewLine +
+                "      Capture 2: 'a'" + Environment.NewLine +
+                "      Capture 3: 'short'" + Environment.NewLine +
+                "      Capture 4: 'sentence'" + Environment.NewLine;
+
+            Assert.Equal(expected, actual.ToString());
+        }
+
+        // https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.capture?view=netcore-3.1#examples
+        [Theory]
+        [InlineData(RegexOptions.None)]
+        [InlineData(RegexOptions.Compiled)]
+        public void Docs_Capture_Sentences(RegexOptions options)
+        {
+            const string Pattern = @"((\w+)[\s.])+";
+            const string Input = "Yes. This dog is very friendly.";
+
+            var actual = new StringBuilder();
+            foreach (Match match in Regex.Matches(Input, Pattern, options))
+            {
+                actual.AppendLine($"Match: {match.Value}");
+                for (int groupCtr = 0; groupCtr < match.Groups.Count; groupCtr++)
+                {
+                    Group group = match.Groups[groupCtr];
+                    actual.AppendLine($"   Group {groupCtr}: {group.Value}");
+                    for (int captureCtr = 0; captureCtr < group.Captures.Count; captureCtr++)
+                    {
+                        actual.AppendLine($"      Capture {captureCtr}: {group.Captures[captureCtr].Value}");
+                    }
+                }
+            }
+
+            string expected =
```

Comment:
Might be easier to read to use \n in the strings, then .Replace at the end to make Environment.Newline if necessary.